### PR TITLE
Check that cloud sql provider version is valid

### PR DESCRIPTION
**Pull request by potiuk**
Additional chek on cloud sql version should be done to avoid downloading non-existing binary.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/google/cloud/hooks/cloud_sql.py [hunk 1]
```diff
@@ -482,6 +475,24 @@ def _download_sql_proxy_if_needed(self) -> None:
...
+    def _get_sql_proxy_download_url(self):
+        system = platform.system().lower()
+        processor = os.uname().machine
+        if processor == "x86_64":
+            processor = "amd64"
+        if not self.sql_proxy_version:
+            download_url = CLOUD_SQL_PROXY_DOWNLOAD_URL.format(system, processor)
+        else:
+            if not CLOUD_SQL_PROXY_VERSION_REGEX.match(self.sql_proxy_version):
+                raise Exception(
```
Comment (ashb) [inline airflow/providers/google/cloud/hooks/cloud_sql.py]: Nit:

```suggestion
                raise ValueError(
```

File: tests/providers/google/cloud/hooks/test_cloud_sql.py [hunk 2]
```diff
@@ -1185,3 +1200,53 @@ def test_hook_with_correct_parameters_mysql_tcp(self, get_connection):
...
+def get_processor():
+    processor = os.uname().machine
+    if processor == "x86_64":
+        processor = "amd64"
+    return processor
+
+
+class TestCloudSqlProxyRunner:
+    @pytest.mark.parametrize(
+        "version,download_url",
```
Comment (ashb) [inline tests/providers/google/cloud/hooks/test_cloud_sql.py]:
```suggestion
        ["version", "download_url"],
```

File: tests/providers/google/cloud/hooks/test_cloud_sql.py [hunk 3]
```diff
@@ -1185,3 +1200,53 @@ def test_hook_with_correct_parameters_mysql_tcp(self, get_connection):
...
+            "\\",
+        ],
+    )
+    def test_cloud_sql_proxy_runner_version_nok(self, version):
+        runner = CloudSqlProxyRunner(
+            path_prefix="12345678",
+            instance_specification="project:us-east-1:instance",
+            sql_proxy_version=version,
+        )
+        with pytest.raises(Exception, match="The sql_proxy_version should match the regular expression"):
```
Comment (ashb) [inline tests/providers/google/cloud/hooks/test_cloud_sql.py]:
```suggestion
        with pytest.raises(ValueError match="The sql_proxy_version should match the regular expression"):
```

**Review by ashb — APPROVED** [review]
Lgtm, just a few nits

**Comment by potiuk** [comment]
> Lgtm, just a few nits

Applied. Tests passed. merging.

**Comment by Elonmucks** [comment]
Wow need more cash

**Pull request merged**